File: backend/src/main/java/it/unical/inf/ea/backend/dto/creation/ProductCreateDTO.java
package it.unical.inf.ea.backend.dto.creation;


import it.unical.inf.ea.backend.data.entities.ProductImage;
import it.unical.inf.ea.backend.dto.*;
import it.unical.inf.ea.backend.dto.enums.ProductAvailability;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.hibernate.validator.constraints.Length;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@ToString
public class ProductCreateDTO {

    @Length(max = 100)
    @NotNull
    private String name;

    @Length(max = 1000)
    private String description;

    @Length(max = 1000)
    private String ingredients;

    @Length(max = 1000)
    private String nutritionalValues;

    @NotNull
    private String weight;

    @NotNull
    private int quantity;

    @NotNull
    private BigDecimal price;

    @NotNull
    private BigDecimal shippingCost;

    @NotNull
    private ProductAvailability productAvailability;

    @NotNull
    private BrandDTO brand;

    @NotNull
    private CategoryDTO category;

    private ProductImageDTO image;

    @NotNull
    private boolean onSale;

    private BigDecimal discountedPrice;
}

File: Progetto-2023-2024-Vecchio-Larocca-Lentini-Mahrous/frontend/app/src/main/java/com/android/frontend/dto/creation/ProductCreateDTO.kt
package com.android.frontend.dto.creation

import com.android.frontend.dto.BrandDTO
import com.android.frontend.dto.CategoryDTO
import com.android.frontend.dto.ProductDTO
import com.android.frontend.dto.ProductImageDTO
import java.math.BigDecimal

data class ProductCreateDTO(
        val name: String ?= null,
        val description: String ?= null,
        val ingredients:String ?= null,
        val nutritionalValues: String ?= null,
        val weight: String ?= null,
        val quantity: Int ?= null,
        val price: BigDecimal ?= null,
        val shippingCost: BigDecimal ?= null,
        val availability: ProductDTO.Availability ?= null,
        val brand: BrandDTO ?= null,
        val category: CategoryDTO ?= null,
        val image: ProductImageDTO ?= null,
        val onSale: Boolean ?= null,
        val salePrice: BigDecimal ?= null
)
